File: code/test_suites/cpp/FunctionBuildTest/src/function_functools_cmp_to_key.cpp
#include <iostream>
#include <map>
#include <vector>
#include <algorithm>
#include <sstream>

class NumberDictManager {
public:
    std::string manage_number_dict(const std::string& operations_str) {
        std::vector<std::string> operations;
        std::stringstream ss(operations_str);
        std::string operation;
        while (std::getline(ss, operation, ';')) {
            operations.push_back(operation);
        }
        std::map<int, int> number_dict;

        auto comparator = [](const std::pair<int, int>& entry1, const std::pair<int, int>& entry2) {
            return entry1.first < entry2.first;
        };

        for (const auto& op : operations) {
            if (op.empty()) {
                continue;
            }
            std::stringstream op_ss(op);
            std::string op_type;
            std::getline(op_ss, op_type, ',');
            if (op_type == "add") {
                int keyAdd, valueAdd;
                op_ss >> keyAdd;
                op_ss.ignore();
                op_ss >> valueAdd;
                number_dict[keyAdd] = valueAdd;
            } else if (op_type == "remove") {
                int keyRemove;
                op_ss >> keyRemove;
                number_dict.erase(keyRemove);
            } else if (op_type == "get") {
                int keyGet;
                op_ss >> keyGet;
                auto it = number_dict.find(keyGet);
                if (it == number_dict.end()) {
                    return "default";
                } else {
                    return std::to_string(it->second);
                }
            } else if (op_type == "sort") {
                std::vector<std::pair<int, int>> sortedItems(number_dict.begin(), number_dict.end());
                std::sort(sortedItems.begin(), sortedItems.end(), comparator);
                std::stringstream sb;
                for (const auto& entry : sortedItems) {
                    if (sb.tellp() != 0) {
                        sb << " ";
                    }
                    sb << entry.first << ":" << entry.second;
                }
                return sb.str();
            } else if (op_type == "sum") {
                int sum = 0;
                for (const auto& entry : number_dict) {
                    sum += entry.second;
                }
                return std::to_string(sum);
            }
        }
        return "";
    }
};

File: code/generate_strategies/IR(summary)/java/java/function_functools_cmp_to_key.java
package com.example;

import java.util.*;
import org.apache.commons.lang3.builder.CompareToBuilder; 

public class NumberDictManager {

    
    public String manage_number_dict(String operations_str) {
        
        String[] operations = operations_str.split(";");
        Map<Integer, Integer> number_dict = new HashMap<>();

        
        Comparator<Map.Entry<Integer, Integer>> comparator = new Comparator<Map.Entry<Integer, Integer>>() {
            @Override
            public int compare(Map.Entry<Integer, Integer> entry1, Map.Entry<Integer, Integer> entry2) {
                return new CompareToBuilder().append(entry1.getKey(), entry2.getKey()).toComparison();
            }
        };

        
        for (String op : operations) {
            
            if (op.trim().isEmpty()) {
                continue;
            }
            String[] op_parts = op.split(",");
            switch (op_parts[0]) {
                case "add":
                    int keyAdd = Integer.parseInt(op_parts[1]);
                    int valueAdd = Integer.parseInt(op_parts[2]);
                    number_dict.put(keyAdd, valueAdd);
                    break;
                case "remove":
                    int keyRemove = Integer.parseInt(op_parts[1]);
                    
                    number_dict.remove(keyRemove);
                    break;
                case "get":
                    int keyGet = Integer.parseInt(op_parts[1]);
                    Integer value = number_dict.get(keyGet);
                    
                    return value == null ? "default" : String.valueOf(value);
                case "sort":
                    
                    List<Map.Entry<Integer, Integer>> sortedItems = new ArrayList<>(number_dict.entrySet());
                    Collections.sort(sortedItems, comparator);
                    StringBuilder sb = new StringBuilder();
                    
                    for (Map.Entry<Integer, Integer> entry : sortedItems) {
                        if (sb.length() != 0) {
                            sb.append(" ");
                        }
                        sb.append(entry.getKey()).append(":").append(entry.getValue());
                    }
                    return sb.toString();
                case "sum":
                    int sum = 0;
                    for (int v : number_dict.values()) {
                        sum += v;
                    }
                    return String.valueOf(sum);
                default:
                    
                    break;
            }
        }
        return "";
    }
}